File: app/src/main/java/srjhlab/com/myownbarcode/Adapter/SelectRecyclerviewAdapter.java
package srjhlab.com.myownbarcode.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import srjhlab.com.myownbarcode.Item.SelectDialogItem;
import srjhlab.com.myownbarcode.R;

public class SelectRecyclerviewAdapter extends RecyclerView.Adapter<SelectRecyclerviewAdapter.ViewHolder> implements View.OnClickListener {
    private final static String TAG = SelectRecyclerviewAdapter.class.getSimpleName();

    private Context mContext = null;
    private List<SelectDialogItem> mItems = null;
    private IClickListener mListener = null;

    public interface IClickListener{
        void onClcik(int id);
    }

    public SelectRecyclerviewAdapter(Context context, List<SelectDialogItem> items, IClickListener listener) {
        Log.d(TAG, "##### SelectRecyclerviewAdapter #####");
        this.mContext = context;
        this.mItems = items;
        this.mListener = listener;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        if (mItems.get(position).getItemType() == SelectDialogItem.INPUT_SELF) {
            holder.mCardView.setBackgroundResource(R.drawable.btn_sel_key_event);
            holder.mCardView.setId(SelectDialogItem.INPUT_SELF);
        } else if (mItems.get(position).getItemType() == SelectDialogItem.INPUT_SCAN) {
            holder.mCardView.setBackgroundResource(R.drawable.btn_sel_scan_event);
            holder.mCardView.setId(SelectDialogItem.INPUT_SCAN);
        } else if (mItems.get(position).getItemType() == SelectDialogItem.INPUT_IMAGE) {
            holder.mCardView.setBackgroundResource(R.drawable.btn_sel_image_event);
            holder.mCardView.setId(SelectDialogItem.INPUT_IMAGE);
        }
    }

    @NonNull
    @Override
    public SelectRecyclerviewAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_select_dialog, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public int getItemCount() {
        return this.mItems.size();
    }

    @Override
    public void onClick(View v) {
        if (v instanceof CardView) {
            switch (v.getId()) {
                case SelectDialogItem.INPUT_SELF:
                    Log.d(TAG, "##### onClick ##### INPUT_SELF");
                    mListener.onClcik(SelectDialogItem.INPUT_SELF);
                    break;
                case SelectDialogItem.INPUT_SCAN:
                    Log.d(TAG, "##### onClick ##### INPUT_SCAN");
                    mListener.onClcik(SelectDialogItem.INPUT_SCAN);
                    break;
                case SelectDialogItem.INPUT_IMAGE:
                    Log.d(TAG, "##### onClick ##### INPUT_IMAGE");
                    mListener.onClcik(SelectDialogItem.INPUT_IMAGE);
                    break;
            }
        }
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private CardView mCardView;

        public ViewHolder(View itemView) {
            super(itemView);
            mCardView = (CardView) itemView.findViewById(R.id.cardview_select_dialog);
            mCardView.setOnClickListener(SelectRecyclerviewAdapter.this);
        }
    }
}

File: app/src/main/kotlin/srjhlab/com/myownbarcode/Adapter/SelectRecyclerViewAdapter.kt
package srjhlab.com.myownbarcode.Adapter

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_select_dialog.view.*
import srjhlab.com.myownbarcode.Item.SelectDialogItem
import srjhlab.com.myownbarcode.R

class SelectRecyclerViewAdapter(items: MutableList<SelectDialogItem>, listener: SelectRecyclerViewAdapter.IClickListener) : RecyclerView.Adapter<SelectRecyclerViewAdapter.ViewHolder>(), View.OnClickListener {
    private val TAG = this.javaClass.simpleName

    private var mItems: MutableList<SelectDialogItem> = items
    private var mListener: IClickListener = listener

    interface IClickListener {
        fun onClick(id: Int)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_select_dialog, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        when (mItems.get(position).itemType) {
            SelectDialogItem.INPUT_SELF -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_input_key)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_SELF
            }
            SelectDialogItem.INPUT_SCAN -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_input_scan)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_SCAN
            }
            SelectDialogItem.INPUT_IMAGE -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_input_image)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_IMAGE
            }
            SelectDialogItem.INPUT_MODIFTY -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_modify_barcode)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_MODIFTY
            }
            SelectDialogItem.INPUT_DELETE -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_delete_barcode)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_DELETE
            }
            SelectDialogItem.INPUT_SHARE -> {
                holder.itemView.textview_select_dialog_item.setText(R.string.string_share_barcode)
                holder.itemView.textview_select_dialog_item.id = SelectDialogItem.INPUT_SHARE
            }
        }
    }

    override fun getItemCount(): Int {
        return this.mItems.size
    }

    override fun onClick(v: View) {
        Log.d(TAG, "##### onCLick #####")
        mListener.onClick(v.id)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        init {
            itemView.textview_select_dialog_item.setOnClickListener(this@SelectRecyclerViewAdapter)
        }

    }
}